plugnpy/metric.py

```diff
@@ -77,31 +100,37 @@ def __init__(
             self.state = 0
 
 
-    def convert_automatic_value(self, value):
+    @staticmethod
+    def convert_automatic_value(value, unit, conversion_factor):
         """Converts values with the right prefix for display."""
-
-        if self.unit != self.BYTES_UNIT and value < 1:
-            keys = ('m', 'u', 'n', 'p')
-        else:
-            keys = ('T', 'G', 'M', 'K')
+        keys = Metric.UNIT_PREFIXES_N if unit != Metric.UNIT_BYTES and value < 1 else Metric.UNIT_PREFIXES_P
 
         for key in keys:
-            multiplication_factor = DISPLAY_UNIT_FACTORS[key](self.conversion_factor)
+            multiplication_factor = Metric.DISPLAY_UNIT_FACTORS[key](conversion_factor)
             if 1.0 / multiplication_factor <= value:
-                return value * multiplication_factor, '{0}{1}'.format(key, self.unit)
-        return value, self.unit
+                return value * multiplication_factor, '{0}{1}'.format(key, unit)
+        return value, unit
 
     def convert_threshold(self, value):
```

Comment:
check if methods like this can be made private? are we intending users to use these themselves